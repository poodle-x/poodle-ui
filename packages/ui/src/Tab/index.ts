import { Tab, TabProps } from "./Tab";
import { StandardThemeConfig } from "../theme";

export * from "./Tab";

export default Tab;

export type TabStyleKeys = "root";

export interface TabThemeConfig
	extends StandardThemeConfig<TabStyleKeys, TabProps> {}
