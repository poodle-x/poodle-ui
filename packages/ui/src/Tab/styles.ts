import { createCSSSystemStandard } from "../theme";
import { CSSSystem } from "../styled";
import { TabProps } from "./Tab";

export function Root(props: TabProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard({
		key: "root",
		config: theme?.Tab,
		props,
		base: [
			{
				minWidth: "100px",
				px: "scale-4",
				py: "scale-2",
				outline: "none",
				border: `2px solid transparent`,
				cursor: "pointer",
				color: "text",
				backGroundColor: "transparent",

				"&:focus": {
					border: `2px dotted`,
					borderColor: "primary.base",
				},

				'&[aria-selected="true"][data-orientation="vertical"]': {
					color: "primary.base",
					borderRight: "2px solid",
					borderRightColor: "primary.base",
				},

				'&[aria-selected="true"][data-orientation="horizontal"]': {
					color: "primary.base",
					borderBottom: "2px solid",
					borderBottomColor: "primary.base",
				},

				"&:disabled": {
					color: "disabledText",
					cursor: "not-allowed",
				},
			},
		],
	});
}
