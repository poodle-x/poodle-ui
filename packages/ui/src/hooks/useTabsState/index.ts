export * from "./useTabsState";
export { default } from "./useTabsState";
