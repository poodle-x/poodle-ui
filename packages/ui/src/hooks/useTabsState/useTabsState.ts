import React, { SetStateAction } from "react";

export type TabValue = string | number;

export enum TabRegisterType {
	TAB = "TAB",
	PANEL = "PANEL",
}

export enum TabOrientation {
	VERTICAL = "vertical",
	HORIZONTAL = "horizontal",
}

export interface TabsContextValue {
	orientation: TabOrientation;
	listTab: [TabValue, string][];
	listTabPanel: [TabValue, string][];
	value?: TabValue;
	onChange?: (
		value: TabValue,
		event: React.SyntheticEvent | KeyboardEvent
	) => void;
	register: (
		type: TabRegisterType,
		value: TabValue,
		id: string,
		index: number
	) => void;
	unregister: (type: TabRegisterType, value: TabValue, id: string) => void;
}

export interface UseTabsStateOptions {
	propValue?: TabValue;
}

export const TabsContext = React.createContext<TabsContextValue>(
	{} as TabsContextValue
);

export default function useTabsState(props: UseTabsStateOptions) {
	const { propValue } = props;
	const [unControlledValue, setUnControlledValue] = React.useState<
		TabValue | undefined
	>();
	const [listTab, setListTab] = React.useState<[TabValue, string][]>([]);
	const [listPanel, setListPanel] = React.useState<[TabValue, string][]>([]);

	const handleRegister: TabsContextValue["register"] = React.useCallback(
		(type, value, id, index) => {
			if (type === TabRegisterType.TAB && index !== undefined) {
				setListTab((lt) => {
					return [...lt.slice(0, index), [value, id], ...lt.slice(index)];
				});
				return;
			}

			if (type === TabRegisterType.PANEL && index !== undefined) {
				setListPanel((lt) => {
					return [...lt.slice(0, index), [value, id], ...lt.slice(index)];
				});
				return;
			}

			const action: SetStateAction<[TabValue, string][]> = (cs) => {
				const current = cs.findIndex((c) => c[0] === value);
				if (current === -1) {
					return [...cs, [value, id]];
				} else {
					return cs.map((c, i) => {
						if (i !== current) {
							return c;
						} else {
							return [value, id];
						}
					});
				}
			};
			if (type === TabRegisterType.TAB) {
				setListTab(action);
			} else if (type === TabRegisterType.PANEL) {
				setListPanel(action);
			}
		},
		[]
	);

	const handleUnregister: TabsContextValue["unregister"] = React.useCallback(
		(type, value) => {
			const action: SetStateAction<[TabValue, string][]> = (cs) => {
				const current = cs.findIndex((c) => c[0] === value);
				if (current !== -1) {
					return cs.reduce<[TabValue, string][]>((prev, cur, index) => {
						if (index !== current) {
							prev.push(cur);
						}
						return prev;
					}, []);
				}
				return cs;
			};
			if (type === TabRegisterType.TAB) {
				setListTab(action);
			} else if (type === TabRegisterType.PANEL) {
				setListPanel(action);
			}
		},
		[]
	);

	React.useEffect(() => {
		if (
			propValue === undefined &&
			unControlledValue === undefined &&
			listTab.length > 0
		) {
			setUnControlledValue(listTab[0][0]);
		}
	}, [propValue, listTab, unControlledValue]);

	return {
		setUnControlledValue,
		value: propValue === undefined ? unControlledValue : propValue,
		isUncontrolled: propValue === undefined,
		listTab,
		listTabPanel: listPanel,
		handleUnregister,
		handleRegister,
	};
}
