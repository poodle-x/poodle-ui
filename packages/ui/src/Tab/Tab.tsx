import React from "react";
import Box, { BoxProps } from "../Box";
import { StandardComponentProps, ThemeConfig } from "../theme";
import useDefaultProps from "../utils/useDefaultProps";
import uSetRef from "../utils/setRef";
import { useClassNames } from "../styled";
import * as styles from "./styles";
import { TabsContext } from "../hooks/useTabsState";
import useTab from "../hooks/useTab";

export interface LocalTabProps {
	children?: React.ReactNode;
	value: string | number;
}

export interface TabProps
	extends Omit<BoxProps, "value">,
		StandardComponentProps,
		LocalTabProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.Tab?.defaultProps;
}

export const Tab: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<TabProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, TabProps>((_props, ref) => {
	const props = useDefaultProps<TabProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const tabsContext = React.useContext(TabsContext);

	const { children, className, value, onClick, id, ...otherProps } = props;

	const { props: htmlProps, setRef } = useTab({
		id,
		value,
	});

	const classes = useClassNames({
		props,
		lists: {
			root: {
				classNames: ["poodle-tab", styles.Root, className],
			},
		},
	});

	function handleClick(e: React.SyntheticEvent) {
		tabsContext.onChange?.(value, e);
		e.persist && e.persist();
		onClick && onClick(e as any);
	}

	const composeRef = React.useCallback(
		(node: HTMLElement) => {
			setRef(node);
			uSetRef(ref, node);
		},
		[setRef, ref]
	);

	return (
		<Box
			data-orientation={tabsContext.orientation}
			as="button"
			{...htmlProps}
			{...otherProps}
			onClick={handleClick}
			className={classes.root}
			ref={composeRef}
		>
			{children}
		</Box>
	);
});

Tab.displayName = "PoodleTab";
