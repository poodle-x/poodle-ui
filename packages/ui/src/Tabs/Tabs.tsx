import React from "react";
import { StandardComponentProps, ThemeConfig } from "../theme";
import useDefaultProps from "../utils/useDefaultProps";
import { useClassNames } from "../styled";
import * as styles from "./styles";
import useTabsState, {
	TabOrientation,
	TabsContext,
	TabsContextValue,
	TabValue,
} from "../hooks/useTabsState";
import Box, { BoxProps } from "../Box";

export { TabOrientation, TabsContext, TabsContextValue, TabValue };

export interface LocalTabsProps {
	/**
	 * Send value to control active tab. By default <Tabs /> component
	 * use internal state to control active tab.
	 */
	value?: TabValue;
	/**
	 * Change tabs horizontal orientation or vertical orientation.
	 * Default is horizontal orientation.
	 */
	orientation?: TabOrientation;
	onChangeValue?: (
		value: TabValue,
		event: React.SyntheticEvent | KeyboardEvent
	) => void;
	children?: React.ReactNode;
}

export interface TabsProps
	extends Omit<BoxProps, "value">,
		StandardComponentProps,
		LocalTabsProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.Tabs?.defaultProps;
}

export const Tabs: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<TabsProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, TabsProps>((_props, ref) => {
	const props = useDefaultProps<TabsProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const {
		children,
		className,
		value,
		onChangeValue,
		orientation,
		...otherProps
	} = props;

	const {
		setUnControlledValue,
		isUncontrolled,
		value: calcValue,
		handleUnregister,
		handleRegister,
		listTabPanel,
		listTab,
	} = useTabsState({ propValue: value });

	const classes = useClassNames({
		props,
		lists: React.useMemo(() => {
			return {
				root: {
					classNames: ["poodle-tabs", styles.Root, className],
				},
			};
		}, [className]),
	});

	const handleChangeValue: TabsContextValue["onChange"] = React.useCallback(
		(v, e) => {
			onChangeValue && onChangeValue(v, e);
			if (isUncontrolled) {
				setUnControlledValue(v);
			}
		},
		[onChangeValue, isUncontrolled, setUnControlledValue]
	);

	return (
		<TabsContext.Provider
			value={{
				listTab,
				listTabPanel,
				value: calcValue,
				register: handleRegister,
				unregister: handleUnregister,
				onChange: handleChangeValue,
				orientation: orientation || TabOrientation.HORIZONTAL,
			}}
		>
			<Box {...otherProps} className={classes.root} ref={ref}>
				{children}
			</Box>
		</TabsContext.Provider>
	);
});

Tabs.displayName = "PoodleTabs";
