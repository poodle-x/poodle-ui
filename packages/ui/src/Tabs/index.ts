import { Tabs, TabsProps } from "./Tabs";
import { StandardThemeConfig } from "../theme";
import { TabValue, TabsContext, TabsContextValue } from "../hooks/useTabsState";

export * from "./Tabs";

export default Tabs;

export type TabsStyleKeys = "root" | "list";

export interface TabsThemeConfig
	extends StandardThemeConfig<TabsStyleKeys, TabsProps> {}

export { TabValue, TabsContext, TabsContextValue };
