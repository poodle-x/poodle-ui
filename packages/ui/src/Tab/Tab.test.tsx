import React from "react";
import { TabOrientation } from "../hooks/useTabsState";
import { Tabs } from "../Tabs";
import { ThemeConfig } from "../theme";
import { render, userEvent } from "../utils/test";
import Tab from "./";

describe("<Tab />", () => {
	it("should render Tab correctly", async () => {
		const { findByRole, rerender } = render(
			<Tabs>
				<Tab value="text">Tab</Tab>
			</Tabs>
		);
		expect(await findByRole("tab")).toMatchSnapshot();

		rerender(
			<Tabs orientation={TabOrientation.VERTICAL}>
				<Tab value="text">Tab</Tab>
			</Tabs>
		);

		expect(await findByRole("tab")).toMatchSnapshot();
	});

	it("should render Tab selected correctly", async () => {
		const { findAllByRole } = render(
			<Tabs value="text2">
				<Tab value="text">Tab</Tab>
				<Tab value="text2">Tab2</Tab>
			</Tabs>
		);
		expect(await findAllByRole("tab")).toMatchSnapshot();
	});

	it("should preserve onClick", async () => {
		const mock = jest.fn();
		const { getByTestId } = render(
			<Tabs>
				<Tab data-testid="test" onClick={mock} value="text">
					Tab
				</Tab>
				<Tab value="text2">Tab2</Tab>
			</Tabs>
		);

		const tab = getByTestId("test");

		userEvent.click(tab);

		expect(mock).toHaveBeenCalled();
	});

	it("ref should work", () => {
		const ref = React.createRef<HTMLDivElement>();

		render(
			<Tabs>
				<Tab ref={ref} value="text">
					Tab
				</Tab>
			</Tabs>
		);
		expect(ref.current).toMatchSnapshot();
	});

	it("should render Box props correctly", async () => {
		const { findByRole } = render(
			<Tabs>
				<Tab id="id" mt="20px" value="text">
					Tab
				</Tab>
			</Tabs>
		);
		expect(await findByRole("tab")).toMatchSnapshot();
	});

	describe("theming", () => {
		const themeTest: ThemeConfig = {
			mode: "mode",
			Tab: {
				styles: {
					root: {
						mt: 10,
						letterSpacing: 250,
					},
				},
				variants: {
					custom: {
						root: {
							top: 0,
							textIndent: 1,
						},
					},
				},
			},
		};

		it("should extend styles in global", async () => {
			const { findByRole } = render(
				<Tabs variant="1">
					<Tab variant="custom" value="text">
						Tab
					</Tab>
				</Tabs>,
				{
					theme: themeTest,
				}
			);
			expect(await findByRole("tab")).toMatchSnapshot();
		});

		it("should extend styles in local", async () => {
			const { findByRole } = render(
				<Tabs>
					<Tab themeExtend={themeTest} value="text">
						Tab
					</Tab>
				</Tabs>
			);
			expect(await findByRole("tab")).toMatchSnapshot();
		});

		it("should render with default props correctly", async () => {
			const { findByRole } = render(
				<Tabs>
					<Tab value="text">Tab</Tab>
				</Tabs>,
				{
					theme: {
						Tab: {
							defaultProps: {
								id: "id",
								className: "hello",
							},
						},
					},
				}
			);
			expect(await findByRole("tab")).toMatchSnapshot();
		});

		it("should override styles", async () => {
			const { findByRole } = render(
				<Tabs>
					<Tab value="text">Tab</Tab>
				</Tabs>,
				{
					theme: {
						Tab: {
							overrides: {
								root: {
									mt: 50,
								},
							},
						},
					},
				}
			);
			expect(await findByRole("tab")).toMatchSnapshot();
		});
	});
});
