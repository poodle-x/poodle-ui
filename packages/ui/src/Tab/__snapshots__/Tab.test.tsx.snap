// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> ref should work 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1 c2"
  data-orientation="horizontal"
  id="poodle-tab-1"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> should render Box props correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

.c2 {
  margin-top: 20px;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1 c2"
  data-orientation="horizontal"
  id="id"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> should render Tab correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1 c2"
  data-orientation="horizontal"
  id="poodle-tab-1"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> should render Tab correctly 2`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1 c2"
  data-orientation="vertical"
  id="poodle-tab-1"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> should render Tab selected correctly 1`] = `
Array [
  .c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<button
    aria-selected="false"
    class="poodle-box c0 poodle-tab c1 c2"
    data-orientation="horizontal"
    id="poodle-tab-1"
    role="tab"
    tabindex="-1"
  >
    Tab
  </button>,
  .c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<button
    aria-selected="true"
    class="poodle-box c0 poodle-tab c1 c2"
    data-orientation="horizontal"
    id="poodle-tab-2"
    role="tab"
    tabindex="0"
  >
    Tab2
  </button>,
]
`;

exports[`<Tab /> theming should disable classes 1`] = `
.c0 {
  box-sizing: border-box;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1"
  data-orientation="horizontal"
  id="poodle-tab-1"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> theming should extend styles in global 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1 c2"
  data-orientation="horizontal"
  id="poodle-tab-1"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> theming should extend styles in local 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
  margin-top: 10px;
  letter-spacing: 250px;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1 c2"
  data-orientation="horizontal"
  id="poodle-tab-1"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> theming should override classes 1`] = `
.c0 {
  box-sizing: border-box;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab morning c1"
  data-orientation="horizontal"
  id="poodle-tab-1"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> theming should override styles 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  margin-top: 50px;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1 c2"
  data-orientation="horizontal"
  id="poodle-tab-1"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;

exports[`<Tab /> theming should render with default props correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  min-width: 100px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  border: 2px solid transparent;
  cursor: pointer;
  color: #282828;
  back-ground-color: transparent;
}

.c1:focus {
  border: 2px dotted;
  border-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="vertical"] {
  color: #00478A;
  border-right: 2px solid;
  border-right-color: #00478A;
}

.c1[aria-selected="true"][data-orientation="horizontal"] {
  color: #00478A;
  border-bottom: 2px solid;
  border-bottom-color: #00478A;
}

.c1:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<button
  aria-selected="true"
  class="poodle-box c0 poodle-tab c1 hello c2"
  data-orientation="horizontal"
  id="id"
  role="tab"
  tabindex="0"
>
  Tab
</button>
`;
